ts-client/src/vault/index.ts

```diff
@@ -74,13 +88,29 @@ export default class VaultImpl implements VaultImplementation {
     opt?: {
       cluster?: Cluster;
       programId?: string;
+      affiliateId?: PublicKey;
+      affiliateProgramId?: string;
     },
   ): Promise<VaultImpl> {
     const provider = new AnchorProvider(connection, {} as any, AnchorProvider.defaultOptions());
     const program = new Program<VaultIdl>(IDL as VaultIdl, opt?.programId || PROGRAM_ID, provider);
 
     const { vaultPda, tokenVaultPda, vaultState, lpSupply } = await getVaultState(tokenInfo, program);
-    return new VaultImpl(program, { tokenInfo, vaultPda, tokenVaultPda, vaultState, lpSupply }, opt);
+    return new VaultImpl(
+      program,
+      { tokenInfo, vaultPda, tokenVaultPda, vaultState, lpSupply },
+      {
+        ...opt,
+        affiliateId: opt?.affiliateId,
+        affiliateProgram: opt?.affiliateId
+          ? new Program<AffiliateVaultIdl>(
+            AffiliateIDL as AffiliateVaultIdl,
+            opt?.affiliateProgramId || AFFILIATE_PROGRAM_ID,
+            provider
+          )
+          : undefined
+      }
+    );
   }
 
   public async getUserBalance(owner: PublicKey): Promise<BN> {
```

Comment:
we need to update user balance function if they are depositing though affiliate program
https://github.com/mercurial-finance/vault-periphery/blob/main/affiliate/rust-client/src/user.rs#L136
`user_state.lp_token`